runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -398,4 +398,12 @@ List<CorfuStoreEntry<K, V, M>> getByIndex(@Nonnull final String indexName,
                         entry.getValue().getMetadata()))
                 .collect(Collectors.toList());
     }
+
+    /**
+     * WARNING: Never call this method for anything other than checkpointing!
+     * @return the underlying CorfuTable only for checkpointing purposes
+     */
+    public CorfuTable getCorfuTableForCheckpointingOnly() {
+        return this.corfuTable;
+    }
```

Comment:
@SravanthiAshokKumar I know you didn't write this, but I would like to point out that there are more effective ways to prevent the applications from accessing this field than implementing a public getter method and annotating it with a comment that says "Warning Never call this method ...". You can use encapsulation to achieve this type of hiding, that way the access is enforced via the compiler and not a source code comment.

Reply:
Good point, will try to see if we can better encapsulate this method.